### PR TITLE
Finished flight

### LINKED ISSUE #363: Finished flight details screen
* Add finished flight to FlightDetail
* When clicking on a finished flight in the stat screen, navigate to flight detail screen and displays all the information
* Add a `add report` button

### PR DESCRIPTION
**Pull request by hacknstack**
Closes #363 . 
Added a finished flight screen in FlightDetail, to be used in the stats screen

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/components/FlightDetail.kt [hunk 1]
```diff
@@ -194,6 +214,40 @@ fun ConfirmFlightTimes(confirmedFlight: ConfirmedFlight, padding: Dp, cardColor:
...
+          "Takeoff time" to finishedFlight.takeOffTime,
+          "Landing time" to finishedFlight.landingTime)
+
+  Card(colors = CardDefaults.cardColors(cardColor)) {
+    LargeTitle(title = "Operational times", padding = padding, color = Color.Black)
+    times.forEach { (label, time) ->
+      DisplaySingleMetric(
+          metric = label, value = DateUtility.localTimeToString(DateUtility.dateToLocalTime(time)))
+    }
+    DisplaySingleMetric("Flight Duration", longToTime(finishedFlight.flightTime))
```
Comment (Prolas) [inline app/src/main/java/ch/epfl/skysync/components/FlightDetail.kt:237]: Why not add this in the `map` ?

File: app/src/main/java/ch/epfl/skysync/components/FlightDetail.kt [hunk 2]
```diff
@@ -209,6 +263,25 @@ fun ConfirmFlightLocation(confirmedFlight: ConfirmedFlight, padding: Dp, cardCol
...
 
+/**
+ * Composable function for displaying the takeoff and landing location of a finished flight.
+ *
+ * @param finishedFlight The finished flight object containing details to be displayed.
+ * @param padding The padding value to be applied around the content.
+ * @param cardColor The color to be used for the card background.
+ */
+@Composable
+fun FinishedFlightLocation(finishedFlight: FinishedFlight, padding: Dp, cardColor: Color) {
```
Comment (Prolas) [inline app/src/main/java/ch/epfl/skysync/components/FlightDetail.kt:274]: This should be 2 separated items. You can use the following function, create 2 items with the right arguments: `"Takeoff location"`, `light.takeOffLocation.name` and `"Landing location"`, `flight.landingLocation.name`
```kotlin
@Composable
fun FinishedFlightLocation(title: String,locationName: String, padding: Dp, cardColor: Color) {
    Card(colors = CardDefaults.cardColors(cardColor), modifier = Modifier.fillMaxWidth()) {
        LargeTitle(title = title, padding = padding, color = Color.Black)
        HyperLinkText(location = locationName, padding = padding)
    }
}
```

File: app/src/main/java/ch/epfl/skysync/components/FlightDetail.kt [hunk 3]
```diff
@@ -194,6 +214,40 @@ fun ConfirmFlightTimes(confirmedFlight: ConfirmedFlight, padding: Dp, cardColor:
...
+    LargeTitle(title = "Operational times", padding = padding, color = Color.Black)
+    times.forEach { (label, time) ->
+      DisplaySingleMetric(
+          metric = label, value = DateUtility.localTimeToString(DateUtility.dateToLocalTime(time)))
+    }
+    DisplaySingleMetric("Flight Duration", longToTime(finishedFlight.flightTime))
+  }
+}
+
+fun longToTime(timeInMillis: Long): String {
```
[the remark this answers is not in the document]
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/components/FlightDetail.kt:241]: Thanks!

File: app/src/main/java/ch/epfl/skysync/components/FlightDetail.kt (hunk 1 above)
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/components/FlightDetail.kt:237]: Did not think to do it like this but it's quite smart

File: app/src/main/java/ch/epfl/skysync/components/FlightDetail.kt (hunk 2 above)
Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/components/FlightDetail.kt:274]: Done

File: app/src/main/java/ch/epfl/skysync/components/FlightDetail.kt [hunk 4]
```diff
@@ -194,6 +226,33 @@ fun ConfirmFlightTimes(confirmedFlight: ConfirmedFlight, padding: Dp, cardColor:
...
+ * @param padding The padding value to be applied around the content.
+ * @param cardColor The color to be used for the card background.
+ */
+@Composable
+fun FinishedFlightTimes(finishedFlight: FinishedFlight, padding: Dp, cardColor: Color) {
+  val times =
+      mapOf(
+          "Takeoff time" to finishedFlight.takeOffTime,
+          "Landing time" to finishedFlight.landingTime,
+          "Flight duration" to
```
Comment (Prolas) [inline app/src/main/java/ch/epfl/skysync/components/FlightDetail.kt:242]: Looks a little bit complicated. It is easier to make a `Map<String, String>` like this
```kotlin
  val times =
      mapOf(
          "Takeoff time" to DateUtility.localTimeToString(DateUtility.dateToLocalTime(finishedFlight.takeOffTime)),
          "Landing time" to DateUtility.localTimeToString(DateUtility.dateToLocalTime(finishedFlight.landingTime)),
          "Flight duration" to formatTime(finishedFlight.flightTime))
```
which simplifies `DisaplayMetric`
```kotlin
DisplaySingleMetric(metric = label, value = time)
```

Comment (hacknstack) [reply app/src/main/java/ch/epfl/skysync/components/FlightDetail.kt:242]: Yes

**Review by Prolas — APPROVED** [review]
LGTM

**Pull request merged**